File: src/main/scala/com/kapilkoju/autopos/catalog/service/ItemService.scala
package com.kapilkoju.autopos.catalog.service

import java.lang.Long

import com.kapilkoju.autopos.catalog.domain.Item
import org.springframework.data.domain.{Page, Pageable}

trait ItemService {

  def save(item: Item): Item

  def findAll(queryTerm: String, pageable: Pageable): Page[Item]

  def getItem(id: Long): Option[Item]

  def getItemWithDetail(id: Long): Option[Item]

  def delete(id: Long)

  def adjustQuantity(item: Item, number: Integer)

  def addQuantity(item: Item, number: Integer) {
    adjustQuantity(item, number)
  }

  def subtractQuantity(item: Item, number: Integer) {
    adjustQuantity(item, -number)
  }
}

File: src/main/kotlin/com/kapilkoju/autopos/catalog/service/ItemService.kt
package com.kapilkoju.autopos.catalog.service

import com.kapilkoju.autopos.catalog.domain.Item
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface ItemService {

    fun save(item: Item): Item

    fun findAll(query: String, pageable: Pageable): Page<Item>

    fun getItem(id: Long): Item?

    fun getItemWithDetail(id: Long): Item?

    fun delete(id: Long)

    fun adjustQuantity(item: Item, number: Int)

    fun addQuantity(item: Item, number: Int) {
        adjustQuantity(item, number)
    }

    fun subtractQuantity(item: Item, number: Int) {
        adjustQuantity(item, -number)
    }
}
